lib/storage/src/content_manager/consensus_manager.rs

```diff
@@ -348,11 +348,15 @@ impl<C: CollectionContainer> ConsensusManager<C> {
             if stop_consensus {
                 return Ok(stop_consensus);
             }
-            self.persistent
-                .write()
-                .entry_applied()
-                .context("Failed to save new state of applied entries queue")?;
+
+            self.persistent.write().entry_applied();
         }
+
+        self.persistent
+            .write()
+            .save()
+            .context("Failed to save new state of applied entries queue")?;
```

Comment:
I think it should be acceptable to only save this state *once*, after applying all entries.

This is a simple operation, that just dumps a JSON on disk. In the worst case, if the node crashed *after* applying some operations, but before persisting the state, it will cause the node to *reapply* some operations. The only issue with reapplying operations, that I can think of, is reapplying delete collection/drop shard operations.

E.g.:

Let's assume we have this WAL:

1. delete collection "OOPS"
2. create collection "OOPS"

We applied *both* operations, but crashed before persisting the change.

So now we will reapply these operations, and so we will delete *a possibly non-empty* collection, and then create *an empty* one in place of it.

However, I still think that this is not an issue *in practice*. The `apply_entries` method (and its loop) is not very hot most of the time. We usually don't create and delete thousands of collections every second, but only apply a *handful* of operations. In most cases the loop will run for 10 or so operations and then persist the state right after.

The window for the scenario above to happen would be tiny. 

Another case is when the node is lagging behind a lot and the loop is very hot and applying hundreds of operations per `apply_entries` call... but if we crash when syncing that is even less of an issue, cause the node have to sync with consensus first, and then its shards would be marked as dead at the end of sync anyway.

Reply:
Also, we can tweak this, by calling `save` after *some* operations that might lose data if reapplied improperly (like delete collection/drop shard/etc), but not all.

Reply:
Also, we can revert this change for this PR and decide what to do with this later. It was just convenient for the demonstration.